pkg/operator/ceph/object/user/controller.go

```diff
@@ -124,6 +129,78 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// watch secrets referenced by CephObjectStoreUser.spec.keys
+	const (
+		secretNameField = "spec.keys.secretNames"
+	)
+
+	err = mgr.GetFieldIndexer().IndexField(context.TODO(), &cephv1.CephObjectStoreUser{TypeMeta: controllerTypeMeta}, secretNameField, func(obj client.Object) []string {
+		var secretNames []string
+		for _, k := range obj.(*cephv1.CephObjectStoreUser).Spec.Keys {
+			if k.AccessKeyRef != nil && k.AccessKeyRef.Name != "" {
+				secretNames = append(secretNames, k.AccessKeyRef.Name)
+			}
+			if k.SecretKeyRef != nil && k.SecretKeyRef.Name != "" {
+				secretNames = append(secretNames, k.SecretKeyRef.Name)
+			}
+		}
+		slices.Sort(secretNames)
+		return slices.Compact(secretNames)
+	})
+	if err != nil {
+		logger.Errorf("failed to setup IndexField for CephObjectStoreUser.Spec.Keys. %v", err)
+		return err
```

Comment:
Log and return vs `return errors.Wrapf()`?

Reply:
Agreed. I'm not sure how it ended up like that.

Reply:
Done.